s3cmd

```diff
@@ -583,12 +594,79 @@ def cmd_object_del(args):
 
         if not recursive:
             rc = subcmd_object_del_uri(uri_str)
-        else:
+        elif Config().exclude:
             rc = subcmd_batch_del(uri_str = uri_str)
+        else:
+            rc = subcmd_batch_del_iterative(uri_str = uri_str)
         if not rc:
             return rc
     return EX_OK
 
+def subcmd_batch_del_iterative(uri_str = None, bucket = None):
```

Comment:
I'll think about that, but I'm not sure it would help.  Both `subcmd_batch_del` and `subcmd_object_del_uri` pass off the `uri_str` to `fetch_remote_list()`, which expects a string not an `S3Uri`.  I'll see if I can come up with something simpler, though.